src/main/java/bgu/spl/mics/application/subscribers/M.java

```diff
@@ -108,5 +106,24 @@ public Report createReport(String missionName, int timeIssued, int MoneypennySer
         report.setQTime(Qtime);
         return report;
     }
+    private AgentsAvailableObject acuireAgents(MissionInfo info, List<String> agentsSerials) {
+        AgentsAvailableObject agentsAvailableObject = new AgentsAvailableObject(agentsSerials, info.getDuration());
+        Future<AgentsAvailableObject> agentsAvailableFuture;
+        agentsAvailableFuture = this.getSimplePublisher().sendEvent(new AgentsAvailableEvent(agentsAvailableObject));
+        // check whether Moneypenny acquired the agents, and allocate all relevant details. timeout is needed for the
+        // case where all Moneypennys already terminated, and M is in the middle of processing a mission.
+        agentsAvailableObject = agentsAvailableFuture.get(); // M and Moneypenny uses the same agentsObject
```

Comment:
Why is this without timeout?

Reply:
it's a mistake. how long do you think the timeout should be?... it could take really a lot of time...
if we are waiting for a mission that takes around, 10 time-ticks, it will be a whole second... we'll wait for 5 seconds?..

Reply:
What if we just resolve all Futures with null upon unregistering? Then just check for a null and use inifnite get

Reply:
when you say it, its exactly what I thought! :laughing: that's what I wrote in the docs